olive/workflows/run/run.py

```diff
@@ -183,3 +192,32 @@ def run(config: Union[str, Path, dict], setup: bool = False, data_root: str = No
             config.engine.evaluate_input_model,
         )
         return best_execution
+
+
+def check_local_ort_installation(ep_mapping: Dict[str, str], package_name: str):
+    # Don't need try except since olive itself cannot be imported and run without onnxruntime
+    # there are onnxruntime imports in many places
+    import onnxruntime as ort
+
+    # check for expected ep as a heuristic to see if the correct ort package is installed
+    if ep_mapping[package_name] in ort.get_available_providers():
+        logger.info(f"onnxruntime installation has {ep_mapping[package_name]} as expected.")
+        return
+
+    # uninstall all ort packages
+    all_packages = list(ep_mapping.keys())
+    all_nightly_packages = []
+    for package in all_packages:
+        nightly_package_parts = ["ort-nightly"]
+        if "-" in package:
+            nightly_package_parts.append(package.split("-")[1])
+        all_nightly_packages.append("-".join(nightly_package_parts))
+    all_packages.extend(all_nightly_packages)
+    logger.info(f"Uninstalling all onnxruntime packages: {all_packages}")
+    subprocess.check_call(["python", "-m", "pip", "uninstall", "-y"] + all_packages)
```

Comment:
I'm okay we install new packages for user, but I'm not sure if we could help user decide to automatically uninstall existing packages for user. If the package is incompatible, I prefer raising an error and giving them a suggestion what to do here.

Reply:
I can comment out the uninstall part but it will probably lead to package clashes. Because of how ort packages all of their different flavors using different pip packages but they all install into the same `onnxruntime` namespace. 
So, we don't know what ort package is already installed.  

Maybe for the ort package we don't do any of the installation and uninstallations? And just print the commands they have to run? 
- uninstall all existing ort packages
- install required ort package

Reply:
Yeah we can explain this in error message that different onnxruntime packages share the same namespace so they have to uninstall all onnxruntime package and only install the one they need.

Reply:
should we raise an error or print the recommanded commands?

Reply:
We can print the recommended commands for user, as we can get all onnxruntime packages installed in his environment: https://github.com/microsoft/Olive/blob/main/olive/engine/packaging/packaging_generator.py#L206-L207 so we know what they should uninstall.

Reply:
Updated to use `importlib_metadata.package_distributions()`. Didn't use `pkg_resources` since it's deprecated and `importlib` is recommended.